pkg/ddl/column.go

```diff
@@ -365,14 +365,14 @@ func updateNewIdxColsNameOffset(changingIdxs []*model.IndexInfo,
 	}
 }
 
-func updateChangingCol(col *model.ColumnInfo) *model.ColumnInfo {
-	col.ChangeStateInfo = nil
+func updateModifyingCols(oldCol, changingCol *model.ColumnInfo) {
+	changingCol.ChangeStateInfo = nil
 	// After changing the column, the column's type is change, so it needs to set OriginDefaultValue back
 	// so that there is no error in getting the default value from OriginDefaultValue.
 	// Besides, nil data that was not backfilled in the "add column" is backfilled after the column is changed.
 	// So it can set OriginDefaultValue to nil.
-	col.OriginDefaultValue = nil
-	return col
+	changingCol.OriginDefaultValue = nil
+	oldCol.ChangeStateInfo = &model.ChangeStateInfo{DependencyColumnOffset: changingCol.Offset}
```

Comment:
This is used for the following scenarios:
non-owner: create table t (a bigint（colID1, public）, `_Col_$a` bigint（colID2, WR）);
owner: create table t (a int（colID2, public）, `_Tomestone_$a` bigint（colID1, WO）);

Do `insert into t value(20)` to owner, and read values on non-owner.